packages/components/molecules/f-rating/src/components/RatingSingleStarVariant.vue

```diff
@@ -0,0 +1,58 @@
+<template>
+    <div
+        :class="$style['c-rating']"
+        data-test-id="rating-component">
+        <div :class="$style['c-rating-stars-icon']">
+            <star-filled-icon
+                v-if="starRating"
+                :class="[
+                    $style['c-rating-star'],
+                    $style['c-rating-star--filled'],
+                    $style[`c-rating-star--${starRatingSize}`]
+                ]" />
+            <star-icon
+                v-else
+                :class="[
+                    $style['c-rating-star'],
+                    $style['c-rating-star--empty'],
+                    $style[`c-rating-star--${starRatingSize}`]
+                ]" />
+        </div>
+    </div>
+</template>
+
+<script>
+import {
+    StarIcon,
+    StarFilledIcon
+} from '@justeattakeaway/pie-icons-vue';
+
+export default {
+    components: {
+        StarFilledIcon,
+        StarIcon
+    },
+
+    props: {
+        starRating: {
+            type: Number,
+            default: 0
+        },
+        starRatingSize: {
+            type: String,
```

Comment:
I think it means `'small'`, `'medium'` or `'large'`? Although taking a look at the constants file it looks like an object, I was expecting to see an array containing those three values

Reply:
Personally I think an array is a bit cleaner and results in some simpler code
`VALID_STAR_RATING_SIZES.includes(value)` rather than `!!VALID_STAR_RATING_SIZES[value]`.

I do understand there might be performance benefits, especially for a page like SERP where this component could appear hundreds of times. This could be a good candidate for benchmarking?

Reply:
Reverted.